### PR TITLE
Time dependent itcz strength diagnostic

### PR DESCRIPTION
**Pull request by oliverwm1**
Add a time-dependent mid-tropospheric mass streamfunction diagnostic, as well as derived diagnostics of the time-dependent ITZ strength (max minus min of streamfunction) and water vapor path averaged in ascent region. These plots are shown in a new "timeseries" holoviews plot on the home page of the report.

Example report [here](https://storage.googleapis.com/vcm-ml-public/argo/2022-09-21t140009-db2ae5280650/index.html) (see bottom of Timeseries section).

Significant internal changes:
- Slight change to HVPlot so that it does not fail on empty HoloMaps (which can happen occur if some diagnostics are missing when generating a report)

- [x] Tests added

Coverage reports (updated automatically):
- test_unit: [83%](https://output.circle-artifacts.com/output/job/3e8fc5e0-9f89-41b2-ade4-bcbf801f5cbc/artifacts/0/tmp/coverage/htmlcov-test_unit/index.html)

### CODE REVIEW

File: workflows/diagnostics/tests/prognostic/test_compute.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import numpy as np
+import xarray as xr
+
+from fv3net.diagnostics.prognostic_run.compute import itcz_edges
+
+
+def test_itcz_edges():
```
Comment (brianhenn) [inline workflows/diagnostics/tests/prognostic/test_compute.py:7]: Hmm maybe add a test case that doesn't just depend on the hardcoded edges of the ITCZ, i.e., has a min or max within it?

File: workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py [hunk 2]
```diff
@@ -526,6 +535,40 @@ def compute_hist_2d_bias(diag_arg: DiagArg):
     return _assign_diagnostic_time_attrs(hist2d_prog, diag_arg.prediction)
 
 
+@registry_3d.register("300_700_time_dependent_zonal_mean")
```
Comment (brianhenn) [inline workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py:538]: If going to the trouble to make compute these (time, lat) fields, can we see them as Hovmullers?

**Review by brianhenn — COMMENTED** [review]
Thanks @oliverwm1 I think these look useful. Had a couple questions about performance and visualizing things.

File: workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py (hunk 2 above)
Comment (oliverwm1) [reply workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py:538]: Good idea.

File: workflows/diagnostics/tests/prognostic/test_compute.py (hunk 1 above)
Comment (oliverwm1) [reply workflows/diagnostics/tests/prognostic/test_compute.py:7]: Sure.

File: workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py [hunk 3]
```diff
@@ -526,6 +535,40 @@ def compute_hist_2d_bias(diag_arg: DiagArg):
...
+@transform.apply(transform.resample_time, "3H", inner_join=True)
+@transform.apply(transform.daily_mean, datetime.timedelta(days=10))
+def time_dependent_mass_streamfunction(diag_arg: DiagArg):
+    logger.info("Computing mid-troposphere averaged mass streamfunction")
+    prognostic, grid = diag_arg.prediction, diag_arg.grid
+    if _is_empty(prognostic):
+        return xr.Dataset()
+    northward_wind = prognostic["northward_wind"]
+    v_zm = vcm.zonal_average_approximate(grid.lat, northward_wind, lat_name="latitude")
+    psi = vcm.mass_streamfunction(v_zm).sel(pressure=slice(30000, 70000))
```
[the remark this answers is not in the document]
Comment (oliverwm1) [reply workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py:550]: Not really. The "derived diagnostics" can handle computing some diagnostics from other ones but I don't want to save the time- and pressure-dependent zonally averaged field.

It's somewhat wasteful compute, although I don't think it'll be **too** noticeable as one additional 3D variable being zonally averaged when there are about 8 being averaged currently: https://github.com/ai2cm/fv3net/blob/5525667b67c67be86038a808b7ec9c78abd00d86/workflows/diagnostics/fv3net/diagnostics/prognostic_run/constants.py#L148-L163

[the remark this answers is not in the document]
Comment (brianhenn) [reply workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py:550]: OK thanks for looking into it.

**Review by brianhenn — APPROVED** [review]
LGTM thanks

**Pull request merged**